spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/InstanceHealthChecker.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.loadbalancer.core;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.client.ServiceInstance;
+
+/**
+ * Provides API for verifying service instance health.
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 2.2.0
+ */
+public interface InstanceHealthChecker {
```

Comment:
We're getting closer to Ribbon with the addition of another interface. This is something I want to actively avoid.

Reply:
Ok, thought that would be easier for people to customise, but can switch it.